src/Compilers/Core/Portable/Symbols/INamedTypeSymbol.cs

```diff
@@ -182,5 +182,10 @@ public interface INamedTypeSymbol : ITypeSymbol
         /// True if the type is serializable (has Serializable metadata flag).
         /// </summary>
         bool IsSerializable { get; }
+
+        /// <summary>
+        /// True if the type is a natively-sized integer.
+        /// </summary>
+        bool IsNativeInt { get; }
```

Comment:
Why this approach for the API shape? I would have imagined this was more a "special kind" sort of thing. Or does this also return true for IntPtrs?

Reply:
And what about pointer types?

Reply:
> Why this approach for the API shape? I would have imagined this was more a "special kind" sort of thing. Or does this also return true for IntPtrs?

Agreed with Jason here. I'm not sure that this API shape is the right approach. This feels like something that could be an extension method on `INamedTypeSymbol` if appropriate, but not on `INamedTypeSymbol` itself.

---
In reply to: [334597667](https://github.com/dotnet/roslyn/pull/39157#discussion_r334597667) [](ancestors = 334597667)

Reply:
Removed this property. The public API will be addressed in a separate PR.

---
In reply to: [352761772](https://github.com/dotnet/roslyn/pull/39157#discussion_r352761772) [](ancestors = 352761772,334597667)